migrations/20200826125123_van_sync_configuration.js

```diff
@@ -0,0 +1,433 @@
+exports.up = function(knex) {
+  return knex.raw(`
+    alter table public.campaign
+      add column external_system_id uuid references public.external_system(id);
+
+    create index campaign_external_system_id_index on campaign(external_system_id);
+
+    create function tg_campaign_check_exteral_system_id()
+      returns trigger as $$
+    declare
+      v_system_organization_id integer;
+    begin
+      select organization_id
+      from public.external_system
+      where id = NEW.external_system_id
+      into v_system_organization_id;
+
+      if NEW.organization_id <> v_system_organization_id
+      then
+        raise exception 'External system referenced by [external_system_id:%] must belong to the same organization!',
+          NEW.external_system_id;
+      end if;
+
+      return NEW;
+    end;
+    $$ language plpgsql;
+
+    create trigger _500_campaign_external_system_id 
+      before insert or update on campaign 
+      for each row execute procedure tg_campaign_check_exteral_system_id();
+
+    alter table public.all_question_response
+      add constraint interaction_step_id_fk foreign key(interaction_step_id) references public.interaction_step(id);
+
+    create unique index all_question_response_value_interaction_step_idx
+      on public.all_question_response(value, interaction_step_id);
```

Comment:
I have a feeling we're going to see some speed improvements from this one elsewhere as well, and its write frequency is pretty small, so big 👍 !